src/binkmovie/bink_patches.c

```diff
@@ -0,0 +1,225 @@
+//
+// Collection of patches that enable Bink movie support.
+//
+// Author: CCHyper
+//
+// NOTE: If you wish to make the Bink video playback interface
+//       exit on failure to play or find the the .BIK, the compile
+//       the project with "BINK_REQUIRED".
+//
+// WARNING: This playback interface assumes the user is playing at
+//          least 800x600, otherwise it will fail gracefully.
+//
+
+#include "macros/patch.h"
+#include "TiberianSun.h"
+#include <stdio.h>
+
+#include "bink_load_dll.h"
+#include "binkmovie.h"
+
+
+//
+// Play a bink movie.
+//
+BOOL __fastcall Play_Movie_As_Bink(char *filename, int theme, bool a3, bool stretch, bool hidden)
+{
+    WWKeyboardClass__Clear(WWKeyboard);
+    WWMouseClass__Hide_Mouse(WWMouse);
+	
+	// Load dll imports if not already loaded.
+	if (!BinkImportsLoaded) {
+		if (!Load_Bink_DLL()) {
+			WWDebug_Printf("Play_Movie_As_Bink failed to load DLL!\n", filename);
+#ifdef BINK_REQUIRED
+			MessageBoxA(MainWindow, "Unable to load BinkW32.dll!\n", "Error!", MB_OK);
+			Emergency_Exit();
+			exit(1);
+#endif
+			return FALSE;
+		}
+	}
+	
+	if (GameOptionsClass_ScreenWidth < 800 || GameOptionsClass_ScreenHeight < 600) {
+		WWDebug_Printf("Resolution must be at least 800x600 to play Bink movies!\n");
+#ifdef BINK_REQUIRED
+		MessageBoxA(MainWindow, "Resolution must be at least 800x600 to play Bink movies!\n", "Error!", MB_OK);
+		Emergency_Exit();
+		exit(1);
+#endif
+		return FALSE;
+	}
+
+	// Only play fullscreen movies in campaign/singleplayer
+    if (SessionType.GameSession != GAME_CAMPAIGN) {
+        return FALSE;
+    }
+
+    if (hidden) {
+        HiddenSurface->vtable->Fill(HiddenSurface, 0);
+        GScreenClass__Do_Blit(1, HiddenSurface, 0);
+    }
+
+	// Prepare the bink movie player.
+	
+	BinkMovie_Create(filename);
+    BinkBreakoutAllowed = TRUE;
+	
+	// Play!
+    BinkMovie_Play();
+
+    if (hidden) {
+        HiddenSurface->vtable->Fill(HiddenSurface, 0);
+        GScreenClass__Do_Blit(1, HiddenSurface, 0);
+    }
+	
+	// Cleanup dll imports.
+//	if (BinkImportsLoaded) {
+//		Unload_Bink_DLL();
+//	}
```

Comment:
If unloading the imports at this point is not needed, then the code for it could be removed.